src/pymorize/global_attributes.py

```diff
@@ -101,16 +101,25 @@ def _parse_variant_label(self, label: str) -> dict:
     def _source_id_related(self, rule: dict) -> dict:
         source_id = rule.get("source_id")
         cv = self.cv["source_id"][source_id]
-        _inst_id = rule.get("institution_id", None)
-        inst_id = cv["institution_id"]
-        if _inst_id:
-            assert _inst_id in inst_id
+        cv_institution_ids = cv["institution_id"]
+        user_institution_id = rule.get("institution_id")
+
+        if user_institution_id:
+            if user_institution_id not in cv_institution_ids:
+                raise ValueError(
+                    f"Institution ID '{user_institution_id}' is not valid. "
+                    f"Allowed values: {cv_institution_ids}"
+                )
+            selected_institution_id = user_institution_id
         else:
-            if len(inst_id) > 1:
+            # No user-provided institution_id; infer it
+            if len(cv_institution_ids) > 1:
```

Comment:
Maybe we can make it abundantly clear here and do something like:

```suggestion
            if isinstance(inst_id, list) and len(cv_institution_ids) > 1:
```
Lower down, you could also do an instance check for list. That would avoid problems with you accidentally getting a string, or something like that.

...just an idea.

Reply:
`cv_institution_ids` is the institution_ids from CMIP6 CVs. To do a `isinstance` check for `list` in other terms mean checking the schema defined in json for this field. This can be done for all the fields but it steers the focus on a different direction. What make sense is to have module that does schema checking but should that be a part of pymorize. Data source schema validation. plz comment

Reply:
@pgierz, please comment :)

Reply:
IMO I agree with @siligam, there is no need to check for list, since we know it's defined as a list in the json files. If I understand your suggestion @siligam correctly, I would not control with a schema the json files provided by CMIP. It would be a nice-to-have but quite some work to write it all down.